src/NetscriptFunctions/CodingContract.ts

```diff
@@ -65,8 +73,26 @@ export function NetscriptCodingContract(): InternalAPI<ICodingContract> {
       const filename = helpers.string(ctx, "filename", _filename);
       const hostname = _hostname ? helpers.string(ctx, "hostname", _hostname) : ctx.workerScript.hostname;
       const contract = getCodingContract(ctx, hostname, filename);
+
       return structuredClone(contract.getData());
     },
+    getContract: (ctx) => (_filename, _hostname?) => {
+      const filename = helpers.string(ctx, "filename", _filename);
+      const hostname = _hostname ? helpers.string(ctx, "hostname", _hostname) : ctx.workerScript.hostname;
+      const server = helpers.getServer(ctx, hostname);
+      const contract = getCodingContract(ctx, hostname, filename);
+      // asserting type here is required, since it is not feasible to properly type getData
+      return {
+        type: contract.type,
+        data: contract.getData(),
+        attempt: (answer: unknown) => {
+          helpers.checkEnvFlags(ctx);
+          return attemptContract(ctx, server, contract, answer);
+        },
+        description: contract.getDescription(),
+        numTriesRemaining: contract.getMaxNumTries() - contract.tries,
```

Comment:
This needs to be a method, not a data field. And thus, it should checkEnvFlags, as well.